docs/control-structures.rst

```diff
@@ -622,11 +633,6 @@ A ``Error(string)`` exception is generated in the following situations:
    indicate failures by returning ``false``.
 #. If you create a contract using the ``new`` keyword but the contract
```

Comment:
This might be a bit confusing. This can panic with resource error, but forwarding doesn't make sense in this context.

Maybe keep it separate under the heading:
"For the following case, there will be a panic"

Reply:
It's contract creation, not memory array creation.

Reply:
I don't see any forwarding in the following code:

```solidity
  contract D {
  }
  contract C {
      function f() public {
          D d = new D();
      }
  }
```

Here is the relevant IR:
```yul
                let _1 := allocateTemporaryMemory()
                let _2 := add(_1, datasize("D_1"))
                if or(gt(_2, 0xffffffffffffffff), lt(_2, _1)) { panic_error_0x41() }
                datacopy(_1, dataoffset("D_1"), datasize("D_1"))
                _2 := abi_encode_tuple__to__fromStack(_2)

                let expr_10_address := create(0, _1, sub(_2, _1))

                releaseTemporaryMemory()
                let vloc_d_6_address := expr_10_address
```

Reply:
Interesting! This seems to be a bug in the new codegen. The old codegen:
```
    tag_7:
      mload(0x40)
      dup1
      swap2
      sub
      swap1
      0x00
      create
      dup1
      iszero
      dup1
      iszero
      tag_9
      jumpi
      returndatasize
      0x00
      dup1
      returndatacopy
      revert(0x00, returndatasize)
```